tests/doc/test-levels.md

```diff
@@ -0,0 +1,150 @@
+exercises [HandleEquivocationEvidence](https://github.com/cosmos/cosmos-sdk/blob/2bec9d2021918650d3938c3ab242f84289daef80/x/evidence/keeper/infraction_test.go#L42) which contains many interactions with the staking
+keeper.
+
+#### Limitations
+
+Setting up a particular input state may be more challenging since the application is
+starting from "first principles".  Some of this may be addressed by good test fixture
+abstractions with testing of their own.  Tests may also be more brittle, and larger
+refactors could impact application initialization in unexpected ways with harder to
+understand errors.  This could also be seen as a benefit, and indeed the SDK's current
+integration tests were helpful in tracking down logic errors during earlier stages
+of app-wiring refactors.
+
+#### Proposal
+
+Maintain existing integration tests, but move them to `/tests/integration`.
+
+To help limit scope and complexity, prefer the smallest possible number of modules in
+application startup possible, i.e. don't depend on simapp.
+
+Maintain a *small* number of integration tests which use simapp directly to start up
+and test application operations.
+
+Integration tests outnumber end to end tests.
+
+### Simulation / generative testing 
+
+Simulations are a special case of integration tests where deterministically random module
+operations are executed against a running simapp.  No assertions are made for the state
+transitions resulting from module operations but any error will halt and fail the
+simulation.
+
+Modules must implement [AppModuleSimulation.WeightedOperations](https://github.com/cosmos/cosmos-sdk/blob/2bec9d2021918650d3938c3ab242f84289daef80/types/module/simulation.go#L31) to define their
+simulation operations.  Note that not all modules implement this which may indicate a
+gap in current simulation test coverage.
+
+Modules not returning simulation operations: 
+
+- `x/auth`
+- `x/capability`
+- `x/evidence`
+- `x/mint`
+- `x/params`
+
+A separate binary, [runsim](https://github.com/cosmos/tools/tree/master/cmd/runsim), is responsible for kicking off some of these tests and
+managing their life cycle.
+
+### Limitations
+
+- [A success](https://github.com/cosmos/cosmos-sdk/runs/7606931983?check_suite_focus=true) may take a long time to run, 7-10 minutes per simulation in CI. 
+- [Timeouts](https://github.com/cosmos/cosmos-sdk/runs/7606932295?check_suite_focus=true) sometimes occur on apparently successes without any indication why.
+- Useful error messages not provided on [failure](https://github.com/cosmos/cosmos-sdk/runs/7606932548?check_suite_focus=true) from CI, requiring a developer to run
+  the simulation locally to reproduce.
+
+#### Proposal
+
+No changes proposed at this time.
```

Comment:
I think it should precise what we should use. Injected keeper using AppConfig, or SimApp, or something else?